components/src/core/app-bar/app-bar.component.ts

```diff
@@ -0,0 +1,173 @@
+import {
+    AfterViewInit,
+    ChangeDetectionStrategy,
+    ChangeDetectorRef,
+    Component,
+    ElementRef,
+    EventEmitter,
+    Input,
+    OnChanges,
+    OnInit,
+    Output,
+    SimpleChanges,
+    ViewChild,
+    ViewEncapsulation,
+} from '@angular/core';
+import { fromEvent, interval } from 'rxjs';
+import { throttle } from 'rxjs/operators';
+
+@Component({
+    selector: 'pxb-app-bar',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    encapsulation: ViewEncapsulation.None,
+    styleUrls: ['./app-bar.component.scss'],
+    template: `
+        <mat-toolbar
+            #pxbAppBar
+            style="background-color: mediumpurple"
```

Comment:
Let's get rid of the purple.   I've added a 'color=primary' so we can get a theme color instead.

It might be a good idea to added a 'color' prop for the angular AppBar API so a user can change it.

Reply:
yeah it would be nice to inherit all of the props from angular material but I know we had problems with that before.